src/tls.c

```diff
@@ -1978,14 +1978,18 @@ int TLSX_ALPN_GetRequest(TLSX* extensions, void** data, word16 *dataSz)
 #define ALPN_FREE_ALL     TLSX_ALPN_FreeAll
 #define ALPN_GET_SIZE     TLSX_ALPN_GetSize
 #define ALPN_WRITE        TLSX_ALPN_Write
-#define ALPN_PARSE        TLSX_ALPN_ParseAndSet
+#define ALPN_PARSE(a, b, c, d, e) (((e) == client_hello ? \
```

Comment:
Is the parse macro the right place to fix this min check? Have you looked a code size impact? Using IDE/GCC-ARM is a good way to gauge code size impact.

Reply:
I don't want to see the macro doing this.
The macro is do either call or not.
Can you generically check? Table with values and look it up?